src/main/java/ch/njol/skript/expressions/ExprName.java

```diff
@@ -50,50 +54,69 @@
+		"\t\t</ul>",
+		"\t</li>",
+		"\t<li><strong>Items</strong>",
+		"\t\t<ul>",
+		"\t\t\t<li><strong>Name and Display Name:</strong> The <em>custom</em> name of the item (not the Minecraft locale name). Can be changed.</li>",
+		"\t\t</ul>",
+		"\t</li>",
+		"\t<li><strong>Inventories</strong>",
+		"\t\t<ul>",
+		"\t\t\t<li><strong>Name and Display Name:</strong> The name/title of the inventory. " +
+			"Changing name of an inventory means opening the same inventory with the same contents but with a different name to its current viewers.</li>",
+		"\t\t</ul>",
+		"\t</li>",
 		"</ul>"})
 @Examples({"on join:",
 		"	player has permission \"name.red\"",
-		"	set the player's display name to \"<red>[admin]<gold>%name of player%\"",
-		"	set the player's tablist name to \"<green>%name of player%\"",
+		"	set the player's display name to \"<red>[admin] <gold>%name of player%\"",
+		"	set the player's tab list name to \"<green>%player's name%\"",
 		"set the name of the player's tool to \"Legendary Sword of Awesomeness\""})
-@Since("1.4.6 (players' name & display name), <i>unknown</i> (player list name), 2.0 (item name), 2.2-dev20 (inventory name)")
+@Since("before 2.1, 2.2-dev20 (inventory name), INSERT VERSION (non-living entity support, changeable inventory name)")
 public class ExprName extends SimplePropertyExpression<Object, String> {
-	
-	private static final boolean inventoryTitles = Skript.methodExists(Inventory.class, "getTitle");
-	
+
 	@Nullable
 	static final MethodHandle TITLE_METHOD;
-	
+
 	static {
 		MethodHandle _METHOD = null;
 		try {
-			_METHOD = MethodHandles.lookup().findVirtual(Inventory.class, "getName", MethodType.methodType(String.class));
-		} catch (IllegalAccessException | NoSuchMethodException ignore) {}
+			_METHOD = MethodHandles.lookup().findVirtual(Inventory.class, "getTitle", MethodType.methodType(String.class));
+		} catch (IllegalAccessException | NoSuchMethodException ignored) {}
 		TITLE_METHOD = _METHOD;
 	}
-	
-	final static int ITEMSTACK = 1, ENTITY = 2, PLAYER = 4, INVENTORY = 8;
-	final static String[] types = {"itemstacks/slots", "livingentities", "players", "inventories"};
-	
-	private static enum NameType {
-		NAME("name", "name[s]", PLAYER | ITEMSTACK | ENTITY | INVENTORY, ITEMSTACK | ENTITY) {
+
+	private final static int ITEM = 1, ENTITY = 2, PLAYER = 4, INVENTORY = 8;
+	final static String[] types = {"itemstacks/slots", "entities", "players", "inventories"};
+
+	private enum NameType {
+		NAME("name", "name[s]", PLAYER | ITEM | ENTITY | INVENTORY, ITEM | ENTITY | INVENTORY) {
 			@Override
 			void set(@Nullable Object o, @Nullable String name) {
 				if (o == null)
 					return;
-				if (o instanceof LivingEntity) {
-					((LivingEntity) o).setCustomName(name);
-					((LivingEntity) o).setRemoveWhenFarAway(false);
+				if (o instanceof Entity) {
+					((Entity) o).setCustomName(name);

```

Comment:
Just to be sure, this works as expected on 1.9.4 right?

Reply:
Asked a person to try it on their 1.8.8 server, works fine.